src/ui/widgets/EmbeddedDisplay/opiParser.ts

```diff
@@ -150,6 +155,19 @@ export function opiParseActions(
     actions: []
   };
 
+  const descriptionToLocation = (action: ElementCompact): string => {
+    const description =
+      (action.description && action.description._text) || undefined;
+    switch (description) {
+      case "Open component screen":
+        return "main";
+      case "Open detail screen":
+        return "details";
+      default:
+        return "details";
+    }
+  };
+
```

Comment:
Is this a good way of mapping the new pages to either the `main` display or the `details` display. I feel like it's a bad way of doing it but can't think of another way.

Reply:
There must be a way that CS-Studio determines where the screens are opening. I think it might be `<mode>` but I'd have to check.

Reply:
Yes, it's messy. You might find `replace` or `mode` props. I think what we can do for now is: if `mode` is not present or is 1, open in main; if `mode` is 3 open in details.

Reply:
Ah good spot, I think that would make sense just looking through now, 3 seems to be for the details pane.